src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -49,9 +51,9 @@ const std::map<std::string, ov::element::Type>& dtype_to_ov_type() {
     return dtype_to_ov_type_mapping;
 }
 
-ov::Tensor tensor_from_pointer(py::array& array, const ov::Shape& shape) {
+ov::Tensor tensor_from_pointer(py::array& array, const ov::Shape& shape, const ov::element::Type& ov_type) {
     bool is_contiguous = C_CONTIGUOUS == (array.flags() & C_CONTIGUOUS);
-    auto type = Common::dtype_to_ov_type().at(py::str(array.dtype()));
+    auto type = (ov_type == ov::element::undefined) ? Common::dtype_to_ov_type().at(py::str(array.dtype())) : ov_type;
```

Comment:
Is it possible to extend `tensor_from_numpy` with `ov_type` as well? I know that this introduce some problems with order of arguments and **actually introduce a mechanism of casting numpy arrays inside Tensor constructor** (This by-product is somehow very interesting as there will be no need to write `ov.Tensor(my_array.astype(np.different_type))`, now it could be done by our runtime `ov.Tensor(my_array, ov.Type.different_type)`. It may require some additional checks inside bindings such as shared memory arrays cannot be casted, but it is generally possible.). Now `u1/u4/i4` types are now only restricted to shared memory mode by usage of second special constructor.

CC @akuporos what do you think?

Reply:
Discussed offline. Ctor Tensor(array, shared_memory=False) will not be extended.